controller/control.go

```diff
@@ -134,9 +135,46 @@ func (c *Controller) UpdateVolStatus() {
 	} else {
 		c.ReadOnly = true
 	}
+	c.RWReplicaCount = rwReplicaCount
 
-	logrus.Infof("Previously Volume RO: %v, Currently: %v,  Total Replicas: %v,  RW replicas: %v, Total backends: %v",
-		prev, c.ReadOnly, len(c.replicas), rwReplicaCount, len(c.backend.backends))
+	logrus.Infof("Previously Volume RO: %v, Currently: %v, Total Replicas: %v, RW replicas: %v, Total backends: %v",
+		prevState, c.ReadOnly, len(c.replicas), rwReplicaCount, len(c.backend.backends))
+}
+
+// UpdateCheckpoint updates c.Checkpoint based on the last snapshot.
+// This function should be called after every removal/addition of replicas.
+// GetLatestSnapshot and SetCheckpoint will error out in cases when one of
+// the replicas go down in which case we should unset c.Checkpoint
+// Checkpoint should be set only when
+// replication factor number of replicas are in RW state
```

Comment:
thanks.. this comment helps.
Did we considered the case of adding a replica or removing a replica and the manual steps for that?

Reply:
While replacing the replicas, the checkpoint for new replica is nil, in which case all the snapshots are synced. On removing the old replica, checkpoint at controller is automatically set to nil.
Manual steps are required when auto deletion is triggered which will be pushed in auto-deletion PR.